src/Particular.LicensingComponent/ThroughputCollector.cs

```diff
@@ -0,0 +1,226 @@
+
+        return await Task.FromResult(connectionTestResults);
+    }
+
+    public async Task UpdateUserIndicatorsOnEndpoints(List<UpdateUserIndicator> userIndicatorUpdates, CancellationToken cancellationToken) =>
+        await dataStore.UpdateUserIndicatorOnEndpoints(userIndicatorUpdates, cancellationToken);
+
+    public async Task<List<string>> GetReportMasks(CancellationToken cancellationToken) => await dataStore.GetReportMasks(cancellationToken);
+    public async Task UpdateReportMasks(List<string> reportMaskUpdates, CancellationToken cancellationToken) => await dataStore.SaveReportMasks(reportMaskUpdates, cancellationToken);
+
+    public async Task<List<EndpointThroughputSummary>> GetThroughputSummary(CancellationToken cancellationToken)
+    {
+        var endpoints = (await dataStore.GetAllEndpoints(false, cancellationToken)).ToList();
+        var queueNames = endpoints.Select(endpoint => endpoint.SanitizedName).Distinct().ToList();
+        var endpointThroughputPerQueue = await dataStore.GetEndpointThroughputByQueueName(queueNames, cancellationToken);
+        var endpointSummaries = new List<EndpointThroughputSummary>();
+
+        //group endpoints by sanitized name - so to group throughput recorded from broker, audit and monitoring
+        foreach (var endpointGroupPerQueue in endpoints.GroupBy(g => g.SanitizedName))
+        {
+            var data = new List<ThroughputData>();
+            if (endpointThroughputPerQueue.TryGetValue(endpointGroupPerQueue.Key, out var tempData))
+            {
+                data.AddRange(tempData);
+            }
+
+            var isKnownEndpoint = IsKnownEndpoint(endpointGroupPerQueue);
+            var endpointSummary = new EndpointThroughputSummary
+            {
+                //want to display the endpoint name to the user if it's different to the sanitized endpoint name
+                Name = endpointGroupPerQueue.FirstOrDefault(endpoint => endpoint.Id.Name != endpoint.SanitizedName)?.Id.Name ?? endpointGroupPerQueue.Key,
+                UserIndicator = UserIndicator(endpointGroupPerQueue) ?? (isKnownEndpoint ? Contracts.UserIndicator.NServiceBusEndpoint.ToString() : string.Empty),
+                IsKnownEndpoint = isKnownEndpoint,
+                MaxDailyThroughput = data.Max()
+            };
+
+            endpointSummaries.Add(endpointSummary);
+        }
+
+        return endpointSummaries;
+    }
+
+    public async Task<ReportGenerationState> GetReportGenerationState(CancellationToken cancellationToken)
+    {
+        var reportGenerationState = new ReportGenerationState(transport)
+        {
+            ReportCanBeGenerated = await dataStore.IsThereThroughputForLastXDays(30, cancellationToken)
+        };
+
+        if (!reportGenerationState.ReportCanBeGenerated)
+        {
+            reportGenerationState.Reason = "24hrs worth of data needs to exist in the last 30 days.";
```

Comment:
I wonder if we can improve on this error?

Reply:
Any suggestions @johnsimons ?

Reply:
For context, this error message is displayed in ServicePulse
<img width="1199" alt="image" src="https://github.com/Particular/ServiceControl/assets/122651/ca3f1820-c713-4ebe-952f-fafa60a56bd5">